enarx-keep-sgx-shim/src/main.rs

```diff
@@ -33,30 +86,18 @@ pub unsafe extern "C" fn entry(
     _r9: u64,
 ) -> ! {
     extern "C" {
-        fn do_syscall(
-            rdi: u64,
-            rsi: u64,
-            rdx: u64,
-            r10: u64,
-            r8: u64,
-            r9: u64,
-            rax: SysCall,
-        ) -> u64;
+        fn jump(rsp: u64, fnc: u64) -> !;
+        fn exit(code: u8) -> !;
     }
 
-    const MSG: &str = "Γειά σου Κόσμε!\n";
-
-    do_syscall(
-        1,
-        MSG.as_ptr() as _,
-        MSG.len() as _,
-        0,
-        0,
-        0,
-        SysCall::WRITE,
-    );
-    do_syscall(0, 0, 0, 0, 0, 0, SysCall::EXIT);
-    panic!()
+    let mut crt0 = [0u8; 4096];
+    let space = random() as usize & 0xf0;
```

Comment:
16-aligned?

Reply:
well `crt0` itself is not aligned... masking random doesn't make it aligned also.

Reply:
I have manually aligned `crt0`. I'm still not sure why the GPF happens. But with an alignment of 32 the GPF disappears. I have filed a debt issue to debug this.